Import WPT css/css-color/parsing as of 072e7fd7

squelart opened the pull request:
#### a6f9f788114d7d6211e9dc7a81c3e1203da6fa92
<pre>
Import WPT css/css-color/parsing as of 072e7fd7
<a href="https://bugs.webkit.org/show_bug.cgi?id=277239">https://bugs.webkit.org/show_bug.cgi?id=277239</a>
<a href="https://rdar.apple.com/132688889">rdar://132688889</a>

Reviewed by Tim Nguyen.

`import-w3c-tests -t web-platform-tests/css/css-color/parsing` as of
Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/072e7fd7d7391bb49370e6100610e529d6c9d2b5">https://github.com/web-platform-tests/wpt/commit/072e7fd7d7391bb49370e6100610e529d6c9d2b5</a>
which includes an export of <a href="https://commits.webkit.org/278696@main.">https://commits.webkit.org/278696@main.</a>

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/w3c-import.log:
web-platform-tests/wpt@be8e91e

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-relative-color.html:
web-platform-tests/wpt@3a36906

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html:
* LayoutTests/TestExpectations:
Skip Debug crash, or expect Release failure in color-valid-color-mix-function.html
because of new assertions, see webkit.org/b/277244

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-relative-color-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-relative-color.html:
Removed two non-conformant lines added by
<a href="https://github.com/web-platform-tests/wpt/commit/395477011aa81f6353f40ae5c9844d51ed661791">https://github.com/web-platform-tests/wpt/commit/395477011aa81f6353f40ae5c9844d51ed661791</a>
See <a href="https://github.com/w3c/csswg-drafts/issues/10328">https://github.com/w3c/csswg-drafts/issues/10328</a>

Canonical link: <a href="https://commits.webkit.org/281642@main">https://commits.webkit.org/281642@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95f8a29c556d1b02393593afcfb752e3a4e814af

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39859 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/13070 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64426 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/11040 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47535 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11271 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48950 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7676 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62534 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/37119 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52407 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29782 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33830 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9645 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9955 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55693 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9931 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/66157 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4441 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9754 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56317 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4460 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52384 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56486 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13467 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3681 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35666 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36748 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37839 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/36493 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig commented on LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-relative-color.html:
```diff
@@ -755,6 +755,10 @@
     fuzzy_test_valid_color(`lch(from var(--mygray) l 30 h)`);
+
+    // Ensure that converting between legacy and modern sRGB color spaces work as expected.
+    fuzzy_test_valid_color(`color(from rebeccapurple srgb r g b)`, `color(srgb 0.4 0.2 0.6)`, 0.01);
```
I think this is incorrect given the accepted terms of https://github.com/w3c/csswg-drafts/issues/10328.

squelart replied:
I wanted to look into this line and the next one, thanks for chiming in!
With your implementation, they do fail. Also, I think they're adequately covered by tests above, so I'll remove them.

nt1m reviewed (approved):
Exporting first would be ideal

squelart committed:
Import WPT css/css-color/parsing as of 072e7fd7
https://bugs.webkit.org/show_bug.cgi?id=277239
rdar://132688889

Reviewed by Tim Nguyen.

`import-w3c-tests -t web-platform-tests/css/css-color/parsing` as of
Upstream commit: https://github.com/web-platform-tests/wpt/commit/072e7fd7d7391bb49370e6100610e529d6c9d2b5
which includes an export of https://commits.webkit.org/278696@main.

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-invalid-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-contrast-function-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-contrast-function.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/w3c-import.log:
web-platform-tests/wpt@be8e91e

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-computed-relative-color.html:
web-platform-tests/wpt@3a36906

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-color-mix-function.html:
* LayoutTests/TestExpectations:
Skip Debug crash, or expect Release failure in color-valid-color-mix-function.html
because of new assertions, see webkit.org/b/277244

* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-relative-color-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-color/parsing/color-valid-relative-color.html:
Removed two non-conformant lines added by
https://github.com/web-platform-tests/wpt/commit/395477011aa81f6353f40ae5c9844d51ed661791
See https://github.com/w3c/csswg-drafts/issues/10328

Canonical link: https://commits.webkit.org/281642@main
